control-plane/agents/src/bin/core/controller/io_engine/v1/translation.rs

```diff
@@ -266,6 +266,7 @@ impl IoEngineToAgent for v1::nexus::Child {
             state_reason: v1::nexus::ChildStateReason::from_i32(self.state_reason)
                 .map(|f| From::from(ExternalType(f)))
                 .unwrap_or(ChildStateReason::Unknown),
+            faulted_at: self.fault_timestamp.clone().map(|d| d.into()),
```

Comment:
```suggestion
            faulted_at: self.fault_timestamp.into_opt(),
``` 
?

Reply:
I don't see any method into_opt() in the Option type. Any reference ?

Reply:
It's a trait we've added: `use stor_port::IntoOption`.

Reply:
To avoid the `prost_wkt_types` dependency:

```suggestion
            faulted_at: self
                .fault_timestamp
                .clone()
                .and_then(|t| std::time::SystemTime::try_from(t).ok())
                .map(chrono::DateTime::from),
```

Reply:
This is building ok with this suggestion, but I'm seeing timestamp value zero for a faulted child. Debugging if this is due to this usage.

Reply:
Then you can try how prost does it, something like this:

```suggestion
            faulted_at: self.fault_timestamp.clone().and_then(|t| {
                t.normalize();
                NaiveDateTime::from_timestamp_opt(t.seconds, t.nanos as u32)            
            ).and_then(|dt| DateTime::from_utc(dt, Utc)),
```

Reply:
@tiagolobocastro  Since the bug was identified in dataplane (https://github.com/openebs/mayastor/pull/1340/), it's working fine as per earlier suggestion(without NaiveDateTime).

`TSDEBUG: Child nvmf://10.1.0.7:8420/nqn.2019-05.io.openebs:c4e7ad8b-56cf-4c51-a26f-d9fce2680d53?uuid=c4e7ad8b-56cf-4c51-a26f-d9fce2680d53, state 3, fault_time 2023-03-28T07:41:38Z`

Reply:
Great! please revert Cargo.lock.

Reply:
Cargo.lock update looks unrelated to previous prost-wkt-types dependency.

Reply:
It is related, why would it change otherwise?